backend/src/main/java/de/neuefische/teddyscoronadiaries/service/WeatherService.java

```diff
@@ -37,35 +32,24 @@ public ProvinceCapitalWeatherData getProvinceCapitalWeatherData(String province)
             throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Province not known");
         }
 
-        String capitalWoeid = provinceData.get().getCapitalWoeid();
-        Optional<WeatherData> weatherDataForProvinceCapital = getWeatherDataForProvinceCapital(capitalWoeid);
-
-        if(weatherDataForProvinceCapital.isEmpty()) {throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Weather API not available");}
-
-        return getProvinceCapitalWeatherDataFromWeatherData(provinceData.get().getCapital(), weatherDataForProvinceCapital.get());
-    }
-
-    public Optional<WeatherData> getWeatherDataForProvinceCapital(String capitalWoeid) {
-        List<WeatherData> weatherData = metaWeatherApiService.getWeatherData(capitalWoeid);
+        String capital = provinceData.get().getCapital();
+        Optional<Weather> weatherData = openWeatherApiService.getWeatherForProvinceCapital(capital);
 
-        String currentDay = currentDateUtils.getCurrentDay();
+        if(weatherData.isEmpty()) {throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Weather API not available");}
 
-        return weatherData.stream().filter(dataSet -> dataSet.getApplicableDate().equals(currentDay)).findAny();
+        return getProvinceCapitalWeatherDataFromWeatherData(provinceData.get().getCapital(), weatherData.get());
     }
 
-    public ProvinceCapitalWeatherData getProvinceCapitalWeatherDataFromWeatherData(String capital, WeatherData data) {
+    public ProvinceCapitalWeatherData getProvinceCapitalWeatherDataFromWeatherData(String capital, Weather data) {
```

Comment:
```suggestion
    public ProvinceCapitalWeatherData getProvinceCapitalWeatherDataFromWeatherData(String capital, Weather weatherData) {
```
`data` is way too generic.